modules/phps/Cards/Goals/GoalTwoKeepers.php

```diff
@@ -0,0 +1,75 @@
+<?php
+namespace Fluxx\Cards\Goals;
+
+use Fluxx\Game\Utils;
+
+class GoalTwoKeepers extends GoalCard
+{
+    public function __construct($cardId, $uniqueId)
+    {
+        parent::__construct($cardId, $uniqueId);
+        $this->keeper1 = -1;
+        $this->keeper2 = -1;
+    }
+
+    public function goalReachedByPlayer()
+    {
+        $winner_id = $this->checkTwoKeepersWin($this->keeper1, $this->keeper2);
+
+        return $winner_id;
+    }
+
+    function checkTwoKeepersWin(
+        $keeper_nbr_A,
+        $keeper_nbr_B,
+        $without_B = false
+    ) {
+        $cards = Utils::getGame()->cards;
+
+        $keeper_A = $cards->getCardsOfTypeInLocation(
+            "keeper",
+            $keeper_nbr_A,
+            "keepers",
+            null
+        );
+        $keeper_B = $cards->getCardsOfTypeInLocation(
+            "keeper",
+            $keeper_nbr_B,
+            "keepers",
+            null
+        );
+
+        if ($without_B) {
```

Comment:
This is such an outlier (it's used only once), I feels it's clearer to move it to the single rule that's using it and have this function be easier to read.

Reply:
yes, good catch, I just copied the code from Julien planning to refactor this also. But then forgot :)